File: frontend/src/api/content-briefs.ts
import api from "@/lib/api"
import { ApiResponse } from "@/types/api"
import { ContentBrief, BriefSlide } from "@/types/content-brief"

export const contentBriefsApi = {
  list: async (orgId: string): Promise<ApiResponse<ContentBrief[]>> => {
    const res = await api.get(`/content-briefs/${orgId}`)
    return res.data
  },

  get: async (briefId: string): Promise<ApiResponse<ContentBrief>> => {
    const res = await api.get(`/content-briefs/detail/${briefId}`)
    return res.data
  },

  create: async (
    orgId: string,
    clientId: string,
    name: string,
    contentType: string,
    platform: string,
    slides: { slide_title: string; brief_text: string; notes?: string }[],
    deadlineDate?: string
  ): Promise<ApiResponse<ContentBrief>> => {
    const res = await api.post("/content-briefs/", {
      organization_id: orgId,
      client_id: clientId,
      name,
      content_type: contentType,
      platform,
      deadline_date: deadlineDate || null,
      slides,
    })
    return res.data
  },

  update: async (
    briefId: string,
    data: {
      name?: string
      content_type?: string
      platform?: string
      deadline_date?: string | null
    }
  ): Promise<ApiResponse<ContentBrief>> => {
    const res = await api.put(`/content-briefs/${briefId}`, data)
    return res.data
  },

  updateStatus: async (
    briefId: string,
    status: string
  ): Promise<ApiResponse<ContentBrief>> => {
    const res = await api.patch(`/content-briefs/${briefId}/status`, { status })
    return res.data
  },

  updateSlide: async (
    slideId: string,
    data: {
      slide_title?: string
      brief_text?: string
      notes?: string | null
    }
  ): Promise<ApiResponse<BriefSlide>> => {
    const res = await api.put(`/content-briefs/slides/${slideId}`, data)
    return res.data
  },

  delete: async (
    briefId: string
  ): Promise<ApiResponse<{ message: string }>> => {
    const res = await api.delete(`/content-briefs/${briefId}`)
    return res.data
  },

  getUpcoming: async (): Promise<ApiResponse<ContentBrief[]>> => {
    const res = await api.get("/content-briefs/upcoming/list")
    return res.data
  },

  stats: async (
    orgId: string
  ): Promise<ApiResponse<{ total: number; active: number }>> => {
    const res = await api.get(`/content-briefs/stats/${orgId}`)
    return res.data
  },
}

File: backend/app/api/v1/content_briefs.py
from typing import List
from uuid import UUID
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_current_user, get_db
from app.models.user import User
from app.schemas.content_brief import (
    BriefSlideResponse,
    ContentBriefResponse,
    CreateBriefRequest,
    UpdateBriefRequest,
    UpdateSlideRequest,
    UpdateStatusRequest,
)
from app.schemas.common import SuccessResponse
from app.services.content_brief_service import ContentBriefService

router = APIRouter()


@router.post("/", response_model=SuccessResponse[ContentBriefResponse])
async def create_brief(
    data: CreateBriefRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).create_brief(data, current_user.id))


@router.get("/{org_id}", response_model=SuccessResponse[List[ContentBriefResponse]])
async def list_briefs(
    org_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).list_briefs(org_id, current_user.id))


@router.get("/detail/{brief_id}", response_model=SuccessResponse[ContentBriefResponse])
async def get_brief(
    brief_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).get_brief(brief_id, current_user.id))


@router.put("/{brief_id}", response_model=SuccessResponse[ContentBriefResponse])
async def update_brief(
    brief_id: UUID,
    data: UpdateBriefRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).update_brief(brief_id, data, current_user.id))


@router.patch("/{brief_id}/status", response_model=SuccessResponse[ContentBriefResponse])
async def update_status(
    brief_id: UUID,
    data: UpdateStatusRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).update_status(brief_id, data.status, current_user.id))


@router.put("/slides/{slide_id}", response_model=SuccessResponse[BriefSlideResponse])
async def update_slide(
    slide_id: UUID,
    data: UpdateSlideRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).update_slide(slide_id, data, current_user.id))


@router.delete("/{brief_id}")
async def delete_brief(
    brief_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    await ContentBriefService(db).delete_brief(brief_id, current_user.id)
    return SuccessResponse(data={"message": "Brief deleted"})


@router.get("/upcoming/list", response_model=SuccessResponse[List[ContentBriefResponse]])
async def get_upcoming(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).get_upcoming(current_user.id))


@router.get("/stats/{org_id}", response_model=SuccessResponse[dict])
async def get_stats(
    org_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return SuccessResponse(data=await ContentBriefService(db).get_stats(org_id, current_user.id))
